File: 9-palindrome-number/palindrome-number.dart
class Solution {
  bool isPalindrome(int x) {
    // Negative numbers are not palindromes
    if (x < 0) return false;
    
    int original = x;
    int reversed = 0;
    
    while (x > 0) {
      // Get the last digit and build reversed number
      int digit = x % 10;
      reversed = reversed * 10 + digit;
      x ~/= 10; // Use integer division
    }
    
    return original == reversed;
  }
}

File: 9-palindrome-number/palindrome-number.rb
# @param {Integer} x
# @return {Boolean}
def is_palindrome(x)
    # Negative numbers are not palindromes
    return false if x < 0
    
    original = x
    reversed = 0
    
    while x > 0
        # Get the last digit and build reversed number
        digit = x % 10
        reversed = reversed * 10 + digit
        x /= 10
    end
    
    original == reversed
end